### ISSUE TITLE
Ipc/pcle/[Pp]article words accepted by the FST not matching with lexical entries in XML

### ISSUE THREAD

**Issue by aarppe** [UAlbertaALTLab/itwewina#73]
When I search with an Independent particle (primarily tagged as +Ipc), e.g. êkwa, êha, mâna, etc., these appear to be analyzed by the FST in that the analysis is presented in the right column.

But then these analyses are not matched with the lexical entries in the dictionary XML.

As a result, for e.g. êkwa the lexical entry is way down the (long) list, since it's not ranked first due to not being recognized as having an FST analysis (which it does). The missing analysis is provided at the end of the list, but indicating as not matching with a lexical entry in the dictionary (which is not correct). A screenshot of the same happening for êha fits on one screen:

![image](https://user-images.githubusercontent.com/15003407/45860329-cddb3380-bd23-11e8-9a46-46d731647e46.png)

As far as I can deduce, this could simply be due to inconsistency either in the dictionary XML, or itwewina.relabel, or in crk.tagset, in referring to these Independent particles as either 'pcle' or 'Pcle', cf.

cat ~/giella/art/dicts/crk/Wolvengrey/crkeng.xml | grep 'l pos' | less
      <l pos="N">acâhkos</l>
      <l pos="N">acâhkos kâ-osôsit</l>
      <l pos="N">acâhkosa kâ-otakohpit</l>
      <l pos="V">acâhkosiwiw</l>
      <l pos="N">acâwêwikamikos</l>
      <l pos="Pcle">acici</l>
      <l pos="Pcle">acici-</l>

[Some older file(s) uses pos='pcle' instead]

For the sake of consistency, I'd prefer using the initial-capital version throughout (especially if that's done for the nouns and verbs, and pronouns) - but this may need to be systematized in some one of the configuration files.

**Comment by eddieantonio** [comment]
There's also something confusing about looking up "êha" in the analyzer:

```sh
echo "êha" | hfst-lookup -q analyser-gt-desc.omnivorous.hfstol
```

```
êha	êha+Ipc+Err/Orth	0.000000
êha	êha+Ipc	0.000000

```

Why does it show up twice, once with non-standard orthography? It's the exact same characters as the lemma!

**Comment by eddieantonio** [comment]
For added clarity, here's what's in `crkeng.xml` for "êha":

```xml
<e src="nêhiyawêwin : itwêwina / Cree : Words">
   <lg>
      <l pos="Pcle">êha</l>
      <lc>IPJ</lc>
      <stem/>
   </lg>
   <mg>
   <tg xml:lang="eng">
       <t pos="Pcle">yes</t>
   </tg>
   </mg>
</e>
```

What's an IPJ?

**Comment by eddieantonio** [comment]
> As far as I can deduce, this could simply be due to inconsistency either in the dictionary XML, or itwewina.relabel, or in crk.tagset, in referring to these Independent particles as either 'pcle' or 'Pcle' [...]

Confirmed; This is exactly what's happening. NDS analyzes "êha" and gets `lemma="êha"` and **`pos="Ipc"`**. Eventually, it calls `lookupLemmaPOS()` which searches `crkeng.xml` for an entry with the lemma text being _exactly_ "êha" and the POS being "Ipc"  (using a case-insensitive match).  However, **the only entry for "êha" in the dictionary has the pos="Pcle", so no match is ever made**.

Action items
------------

 - [x] determine whether to make change in FST or dictionary (@aarppe)
 - [x] Change POS of particles to "Ipc" of the XML dictionary so that both sources are consistent


---

Traceback attached for my own debugging sake! (most recent call first)

```
lookupLemmaPOS, lexicon.py:305
lookup, lexicon.py:691
decorate, lexicon.py:69
decorate, lexicon.py:77
lookup, morpho_lexicon.py:212
decorate, morpho_lexicon.py:22
do_search_to_obj, search.py:536
search_to_context, search.py:669
get, search.py:848
dispatch_request, views.py:149
view, views.py:84
dispatch_request, app.py:1461
[...library code...]
```

**Comment by aarppe** [comment]
Re: IPJ - In Arok's nomenclature, all other words except the inflecting nouns, verbs, numerals and pronouns are (independent) Particles, under which he has various subclasses.

IPN  : Adjectival/adverbial pre-nouns
IPH:  Multiword particles / fixed phrases
INM : Proper/Place names
IPC : The default class of uninflecting independent particles (translatable as adverbs)
IPJ : Interjections 
IPP : preposition-like pre-verbs/pre-nouns (only 4, not sure)
IPV : Pre-verbs

Will look into the other matters, most likely changing the XML coding rather than the FST.

**Comment by aarppe** [comment]
Changed 'Pcle' to 'Ipc' in the <l> field directly in the source XML files, crkeng.xml and crkeng_ranked.xml, and checked these in SVN.

Changed the script ./wolvengrey2xml.sh accordingly, and checked in SVN.

NOTE: While changing the part-of-speech (pos attribute) for the crk lexical entry presents no issues, I am not entirely sure what on the longer term would be the best solution for setting the pos attribute for the (English) translation, as the parts-of-speech for English are not the same as for Cree, and in too many cases there would not be a satisfyingly 1-to-1 mapping. But this is an issue to worry about later, if even then.

**Comment by eddieantonio** [comment]
This is what it looks like now:

![captura de pantalla 2018-09-25 a las 2 33 03 p m](https://user-images.githubusercontent.com/2294397/46041413-f53d5200-c0cf-11e8-928f-3905afc116c1.png)

I'm calling this one closed!

**Comment by aarppe** [comment]
Ok, yes, this Ipc issues is closed, but it spawned #75 on the +Err/Orth tags as an (accidental) follow-up.
